File: Code Boiler/Java/SignUp.java
package signup.Java;

import java.util.Scanner; 
import java.security.SecureRandom; 


class SignUp {
    private String firstName; 
    private String lastName; 
    private String email; 
    private String password; 
    private String accountNumber; 



    public SignUp() {
        Scanner input = new Scanner(System.in); 


        System.out.print("Enter your first name: "); 
        firstName = input.nextLine(); 
        System.out.print("Enter your last name: "); 
        lastName = input.nextLine(); 
        System.out.print("Enter your email: "); 
        email = input.nextLine(); 
        System.out.print("Enter your password: "); 
        password = input.nextLine(); 

        SecureRandom random = new SecureRandom(); 

        accountNumber = firstName.substring(0, 1) + lastName.substring(0, 1) + Integer.toHexString(random.nextInt()); 
    }


    private boolean validateInput() {
        if (!email.endsWith("@example.com")) {
            System.out.println("Invalid email address"); 
            return false; 
        }
        if (password.length() < 8) {
            System.out.println("Password is short"); 
            return false; 
        }
        return true; 
    }

    public void register() {
        if (!validateInput()) {
            return; 
        }
        System.out.println("Registeration successful! Your account number is " + accountNumber);
        System.exit(0); 
    }

    public static void main(String[] args) {
        SignUp user = new SignUp(); 
        user.register(); 
    }
}

File: Code Boiler/golang/SignUp.go
package main

import (
	"crypto/rand"
	"encoding/hex"
	"fmt"
	"os"
	"strings"
)

type SignUp struct {
	firstName     string
	lastName      string
	email         string
	password      string
	accountNumber string
}

func (s *SignUp) validateInput() bool {
	if !strings.HasSuffix(s.email, "@example.com") {
		fmt.Println("Invalid email address")
		return false
	}
	if len(s.password) < 8 {
		fmt.Println("Password too short")
		return false
	}
	return true
}

func (s *SignUp) register() {
	if !s.validateInput() {
		return
	}
	randomBytes := make([]byte, 4)
	_, err := rand.Read(randomBytes)
	if err != nil {
		fmt.Println("Error generating random bytes:", err)
		os.Exit(1)
	}
	accountNumber := s.firstName[0:1] + s.lastName[0:1] + hex.EncodeToString(randomBytes)
	fmt.Println("Registration successful! Your account number is: ", accountNumber)
	os.Exit(0)
}

func main() {
	user := SignUp{
		firstName: input("Enter your first name: "),
		lastName:  input("Enter your last name: "),
		password:  input("Enter your password: "),
	}
	user.register()
}

func input(prompt string) string {
	fmt.Print(prompt)
	var input string
	fmt.Scanln(&input)
	return input

}
